### ISSUE TITLE
Server Carry Forward Script value is overwritten (to 0) by the client SOH sync when there is no change in value on the client

### ISSUE THREAD

**Issue by chai-lmis** [clintonhealthaccess/chailmis-android#269]

**Comment by ihassin** [comment]
It may be that this happens when the sync fails and zero is stored on client that is in turn echoed back to the server and zeros it out. To debug, we'd need to cooperate with back-end people. We also have to decide how to handle sync issues, and null values reaching the server.
@egwuma - how do you want to handle these kinds of issues and collaboration challenges?

**Comment by chai-lmis** [comment]
i dont believe this is the case as the test cases i had neither had zero on the client or on the server. However when sync fails, the last carried forward value should remain on the server

**Comment by garymabin** [comment]
@chai-lmis. I think we only syncing zero value to the server in one situation, that is we don't get right value from the server side previously. So to make further investigation possible, we need to reproduce the issue and check the server log just at that time and track the latest syncing and latest retrieving log before that issue occurs.

**Comment by ihassin** [comment]
@bin - did you get access to the server logs from Jerome?
Once logged on, you can tail the logs by:
tail -F /home/dhis/tomcat-dhis/logs/catalina.out
